### PR TITLE
add mocks and basic tests

### PR DESCRIPTION
**Pull request by philbrookes**

### CODE REVIEW

File: pkg/controller/installation/installation_controller.go [hunk 1]
```diff
@@ -140,6 +140,7 @@ func (r *ReconcileInstallation) Reconcile(request reconcile.Request) (reconcile.
 				log.Info("Error updating Installation resource. Requeue and retry.")
 				return reconcile.Result{
 					Requeue: true,
+					RequeueAfter: time.Second * 10,
```
Comment (maleck13) [inline pkg/controller/installation/installation_controller.go]: seems arbitrary?

File: pkg/controller/installation/installation_controller.go [hunk 2]
```diff
@@ -154,6 +155,7 @@ func (r *ReconcileInstallation) Reconcile(request reconcile.Request) (reconcile.
...
```
Comment (maleck13) [inline pkg/controller/installation/installation_controller.go]: If this is used every where should be a constant. Is this needed for test reasons?

File: pkg/controller/installation/installation_controller.go (hunk 1 above)
Comment (philbrookes) [reply pkg/controller/installation/installation_controller.go]: yeah, we can change it, it just has to have a value, or else it does backoff logic, and eventually doesn't requeue it for ages.

File: pkg/controller/installation/installation_controller.go (hunk 2 above)
Comment (philbrookes) [reply pkg/controller/installation/installation_controller.go]: as above, it just has to have a value. It's unrelated to tests, just noticed it while poking around with @sedroche

**Pull request merged**